src/bindings/python/tests/test_runtime/test_infer_request.py

```diff
@@ -493,6 +493,17 @@ def callback(request, _):
     infer_queue.wait_all()
 
 
+def test_infer_queue_iteration(device):
```

Comment:
Also the test for `iter(asyncinferqueue_obj)` would be great. It might be added in this one as an assert.